### PR TITLE
Emoji verification: avoid emoji text overlapping

### PR DESCRIPTION
**Pull request by bwindels**
while maintaining some sort of olypmic rings formation.

This makes the emoji columns not a fixed width anymore so the text can fit, at the cost of not maintaining a perfect symmetrical olympic ring formation.

olympic
This is what the dialog looks like
image

Related to https://github.com/vector-im/riot-web/issues/12133

### CODE REVIEW

**Comment by t3chguy** [comment]
can you do a gif with the longest german emoji word (for every emoji) we found?

`Schraubenschlüssel`

File: src/components/views/verification/VerificationShowSas.js [hunk 1]
```diff
@@ -62,9 +62,12 @@ export default class VerificationShowSas extends React.Component {
                 </div>,
             );
             sasDisplay = <div className="mx_VerificationShowSas_emojiSas">
-                {emojiBlocks.slice(0, 4)}
-                <div className="mx_VerificationShowSas_emojiSas_break" />
+                <div className="mx_VerificationShowSas_emojiSas_row">
+                    {emojiBlocks.slice(0, 4)}
+                </div>
+                <div className="mx_VerificationShowSas_emojiSas_row">
                 {emojiBlocks.slice(4)}
```
Comment (t3chguy) [inline src/components/views/verification/VerificationShowSas.js]: indentation

**Comment by bwindels** [comment]
> can you do a gif with the longest german emoji word (for every emoji) we found?
> 
> `Schraubenschlüssel`

image

**Comment by bwindels** [comment]
Or even a GIF: olympic2

**Comment by t3chguy** [comment]
What would happen if all of them were the long emoji word (unlucky) and we had a narrow UserInfo, its an edge case but possible

**Comment by bwindels** [comment]
> What would happen if all of them were the long emoji word (unlucky) and we had a narrow UserInfo, its an edge case but possible

Hmm, there's an overflow: hidden in there, so it would get cropped :-1:

**Comment by jryans** [comment]
I feel like it looks better visually when we keep them closely packed no matter the width, and in a balanced 4 / 3 layout, like this design:

2020-01-30 at 20 17

How about we trim long names with an ellipsis? The emoji alone should be enough to get the idea, and you'll at least get the beginning of the word for the rare cases where they are long. So a few points that would be great to achieve:

* Trim long text with ellipses to avoid overlap
* Keep balanced 4 / 3 layout at all times
* Stay closely packed at any width
* Emoji in modal version are too large, should use font-size 32px there
* Emoji labels should be font-size 12px, normal weight

**Comment by bwindels** [comment]
Sgtm, so we can abandon this PR then. @t3chguy, would you have some time to implement the above suggestions? That I can help Travis out with QR stuff.

**Comment by jryans** [comment]
@t3chguy, let me know if you have time, otherwise I'll try to do something here.

**Pull request closed without merging**